### PR TITLE
Add prediction script and accordion ui

### PR DESCRIPTION
**Pull request by SalnikovIgor**

### CODE REVIEW

File: vector_db/app/database.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 
 class DataBase:
     def __init__(self, settings: dict) -> None:
-        chroma_client = chromadb.PersistentClient()
+        chroma_client = chromadb.PersistentClient(path='C:\\Users\\admin\\arxiv-ai-search\\vector_db\\chroma')
```
Comment (demid5111) [inline vector_db/app/database.py]: .

File: vector_db/app/database.py [hunk 2]
```diff
@@ -21,8 +21,14 @@ def add(self, documents: list[str], embeddings: list[float]) -> None:
...
-            query_texts=[query],
+            query_texts=[text],
             n_results=5
+        )
+    
+    def query_embedding(self, embedding: list[str], n_results=10):
+        return self.collection.query(
+            query_embeddings=embedding.tolist(),
+            n_results=n_results
         )
```
Comment (demid5111) [inline vector_db/app/database.py]: newline

File: vector_db/scripts/prediction.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    full_row = df_full.loc[df_full['_url'] == row]
+    abstract = full_row['_abstract'].str.replace('Abstract:', '').values[0]
+    embedding = model.embedding(abstract)
+    results = db.query_embedding(embedding)
+
+    df.loc[len(df)] = [full_row['_url'].values[0]] + [item['url'] for item in results['metadatas'][0]]
+
+df.to_csv('result.csv')
+
+
```
Comment (demid5111) [inline vector_db/scripts/prediction.py]: nl

File: vector_db/static/js/main.js [hunk 4]
```diff
@@ -35,12 +36,66 @@ searchBtn.addEventListener('click', async () => {
...
     console.log(result);
-    resultsBox.replaceChildren();
-    for (const [index, value] of result['distances'][0].entries()) {
-        console.log(index)
-        li = document.createElement('li');
-        li.classList.add('list-group-item');
-        li.innerHTML = `${value} ${result['documents'][0][index]}`;
-        resultsBox.appendChild(li);
-    }
+    // resultsBox.replaceChildren();
```
Comment (demid5111) [inline vector_db/static/js/main.js]: rm

File: vector_db/templates/index.html [hunk 5]
```diff
@@ -19,9 +19,9 @@
...
             </div>
         </form>
-        <ul class="text list-group" id="results">
-        </ul>
+        <div class="accordion" id="accordion"></div>
     </div>
+    <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.0.2/dist/js/bootstrap.bundle.min.js" integrity="sha384-MrcW6ZMFYlzcLA8Nl+NtUVF0sA7MsXsP1UyJoMp4YLEuNSfAP+JcXn/tWtIaxVXM" crossorigin="anonymous"></script>
     <script src="{{ url_for('static', path='/js/main.js') }}"></script>
 </body>
 </html>
```
Comment (demid5111) [inline vector_db/templates/index.html]: nl

File: result.csv
Comment (demid5111) [inline result.csv:1]: rm from git, put in dist and ignore it

File: vector_db/scripts/prediction.py [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+columns = ['query_article_link']
+for i in range(1, 11):
+    columns.append(f'top{i}')
+df = pd.DataFrame(columns=columns)
+
+for row in df_google['query_article_link']:
+    full_row = df_full.loc[df_full['_url'] == row]
+    abstract = full_row['_abstract'].str.replace('Abstract:', '').values[0]
+    embedding = model.embedding(abstract)
```
Comment (demid5111) [inline vector_db/scripts/prediction.py:40]: should be done via chroma

File: vector_db/scripts/prediction.py [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+df_full = pd.read_csv(args.path_to_full)
+
+
+columns = ['query_article_link']
+for i in range(1, 11):
+    columns.append(f'top{i}')
+df = pd.DataFrame(columns=columns)
+
+for row in df_google['query_article_link']:
```
Comment (demid5111) [inline vector_db/scripts/prediction.py]: should be easy to iterate through rows

File: vector_db/scripts/prediction.py [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+model = Model()
+
+settings = {
+    "name":"arxiv"
+}
+db = DataBase(settings)
+
+df_google = pd.read_csv(args.path_to_target)
+df_google = df_google.dropna()
```
Comment (demid5111) [inline vector_db/scripts/prediction.py:24]: we also drop items that include non-indexed papers

File: vector_db/scripts/prediction.py [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+columns = ['query_article_link']
+for i in range(1, 11):
+    columns.append(f'top{i}')
+df = pd.DataFrame(columns=columns)
+
+for row in df_google['query_article_link']:
+    full_row = df_full.loc[df_full['_url'] == row]
+    abstract = full_row['_abstract'].str.replace('Abstract:', '').values[0]
+    embedding = model.embedding(abstract)
+    results = db.query_embedding(embedding)
```
Comment (demid5111) [inline vector_db/scripts/prediction.py:41]: we make cosine similarity search?

File: vector_db/scripts/prediction.py [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+
+    df.loc[len(df)] = [full_row['_url'].values[0]] + [item['url'] for item in results['metadatas'][0]]
```
Comment (demid5111) [inline vector_db/scripts/prediction.py]: use append

**Review by demid5111 — CHANGES_REQUESTED** [review]

File: vector_db/scripts/prediction.py (hunk 6 above)
Comment (SalnikovIgor) [reply vector_db/scripts/prediction.py:40]: Chroma use  only `all-MiniLM-L6-v2`

File: vector_db/charts/all_years.png
Comment (demid5111) [inline vector_db/charts/all_years.png:1]: rm

File: vector_db/README.md [hunk 11]
```diff
@@ -6,4 +6,14 @@
...
+
+## Results
+
+| Model                      | Metric | Accuracy |
+| -------------------------- | ------ | -------- |
+| all-MiniLM-L6-v2           | Cosin  | 0.013    |
+| all-mpnet-base-v2          | Cosin  |          |
+| all-distilroberta-v1       | Cosin  |          |
+| multi-qa-distilbert-cos-v1 | Cosin  |          |
+
```
Comment (demid5111) [inline vector_db/README.md:19]: 4?

**Review by demid5111 — CHANGES_REQUESTED** [review]

**Pull request merged**